collect_app/src/main/res/raw/settings_schema.json

```diff
@@ -0,0 +1,340 @@
+          "type": "string"
+        },
+        "metadata_phonenumber": {
+          "type": "string"
+        },
+        "metadata_email": {
+          "type": "string"
+        },
+        "background_location": {
+          "type": "boolean"
+        },
+        "background_recording": {
+          "type": "boolean"
+        }
+      }
+    },
+    "admin": {
+      "type": "object",
+      "properties": {
+        "admin_pw": {
+          "type": "string"
+        },
+        "edit_saved": {
+          "type": "boolean"
+        },
+        "send_finalized": {
+          "type": "boolean"
+        },
+        "view_sent": {
+          "type": "boolean"
+        },
+        "get_blank": {
+          "type": "boolean"
+        },
+        "delete_saved": {
+          "type": "boolean"
+        },
+        "change_server": {
+          "type": "boolean"
+        },
+        "change_project_display": {
+          "type": "boolean"
+        },
+        "change_app_theme": {
+          "type": "boolean"
+        },
+        "change_app_language": {
+          "type": "boolean"
+        },
+        "change_font_size": {
+          "type": "boolean"
+        },
+        "change_navigation": {
+          "type": "boolean"
+        },
+        "show_splash_screen": {
+          "type": "boolean"
+        },
+        "maps": {
+          "type": "boolean"
+        },
+        "form_update_mode": {
+          "type": "boolean"
+        },
+        "periodic_form_updates_check": {
+          "type": "boolean"
+        },
+        "automatic_update": {
+          "type": "boolean"
+        },
+        "hide_old_form_versions": {
+          "type": "boolean"
+        },
+        "change_autosend": {
+          "type": "boolean"
+        },
+        "delete_after_send": {
+          "type": "boolean"
+        },
+        "default_to_finalized": {
+          "type": "boolean"
+        },
+        "change_constraint_behavior": {
+          "type": "boolean"
+        },
+        "high_resolution": {
+          "type": "boolean"
+        },
+        "image_size": {
+          "type": "boolean"
+        },
+        "guidance_hint": {
+          "type": "boolean"
+        },
+        "external_app_recording": {
+          "type": "boolean"
+        },
+        "instance_form_sync": {
+          "type": "boolean"
+        },
+        "change_form_metadata": {
+          "type": "boolean"
+        },
+        "analytics": {
+          "type": "boolean"
+        },
+        "moving_backwards": {
+          "type": "boolean"
+        },
+        "access_settings": {
+          "type": "boolean"
+        },
+        "change_language": {
+          "type": "boolean"
+        },
+        "jump_to": {
+          "type": "boolean"
+        },
+        "save_mid": {
+          "type": "boolean"
+        },
+        "save_as": {
+          "type": "boolean"
+        },
+        "mark_as_finalized": {
+          "type": "boolean"
+        }
+      }
```

Comment:
What about `allow_other_ways_of_editing_form` It is not a real setting in our preferences but it's a result of other settings so probably if we do not allow to include it it might lead to some strange states?

Reply:
but including it also might lead to an incorrect state if  the values doesn't match related prefs. So it shouldn't be here but we should make sure we handle this case during importing settings, I'm afraid we don't...

Reply:
I'm actually not familiar with that setting, but I can look into it. So right now could someone include that in JSON and end up in a weird state?

Reply:
That option is set to false if you uncheck `Moving backwards` (in Protected -> Access Control -> From Entry Settings) and click Yes (it has an additional dialog). Then if you check `Moving backwards` it is set to true.

If it's false `Jump to` and `Save` options in form entry are disabled.

So I think if someone loads settings where `Moving backwards` is checked (true) but `allow_other_ways_of_editing_form` is false it leads to a state that normally should not be possible to take place.

Reply:
Interesting. That's hopefully something we can guard against with this schema. Are you able to create an issue for that? It sounds like it's an existing issue unless I'm misunderstanding something.

Reply:
Ok I will report a separate issue for that.